components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/utils/ServiceCatalogUtils.java

```diff
@@ -524,12 +617,15 @@ public static boolean checkPreConditions() {
             log.info("Cannot find carbon applications - aborting the service-catalog uploader");
             return false;
         }
-        // Skip if no APIs are deployed
+        // Skip if no APIs or Proxy Services are deployed
         Collection APITable =
                 SynapseConfigUtils.getSynapseConfiguration(
                         org.wso2.micro.core.Constants.SUPER_TENANT_DOMAIN_NAME).getAPIs();
-        if (APITable.isEmpty()) {
-            log.info("Cannot find APIs - aborting the service-catalog uploader");
+        Collection proxyTable =
+                SynapseConfigUtils.getSynapseConfiguration(
+                        org.wso2.micro.core.Constants.SUPER_TENANT_DOMAIN_NAME).getProxyServices();
+        if (APITable.isEmpty() && proxyTable.isEmpty()) {
+            log.info("Cannot find APIs or Proxy Services - aborting the service-catalog uploader");
```

Comment:
This could probably be a debug log since this is not something significant.

Reply:
Btw, this log prints when the developer has configured Service catalogue configurations in the deployment.toml file. In the 4.0.0 release, we had this log for APIs. WDYT?

Reply:
This will get printed for each task run until the first API is deployed right? Do you see any useful information?

Reply:
No, this is a one-time log. +1 to change this as debug log. Fixed

Reply:
Fixed